toolchain/check/member_access.cpp

```diff
@@ -545,11 +539,13 @@ static auto PerformActionHelper(Context& context, SemIR::LocId loc_id,
                           SemIR::NameId);
         context.emitter().Emit(loc_id, QualifiedExprNameNotFound, base_id,
                                name_id);
+        return SemIR::ErrorInst::SingletonInstId;
+      } else {
+        return SemIR::InstId::None;
```

Comment:
How does this interact with delayed instantiation of `AccessOptionalMemberAction`? I'm worried that if the template action unexpectedly produces a `None` instruction, the consumers of the `InstValue` will crash. Eg:
```carbon
class C(template T:! type) {}
fn F(template T:! type) {
  var v: C(T);
}
fn G() { F({}); }
```
(This example probably fails for various other reasons too, though.)

Reply:
No matter what I do here, even returning a new instruction, I get a crash:

```
CHECK failure at ./toolchain/base/value_store.h:73: id.index >= 0: inst-5
```

inside:

```
#18 0x00005596ecddaa9a RefineOperands /proc/self/cwd/toolchain/check/action.cpp:142:1
#19 0x00005596ecddaa9a Carbon::Check::AddDependentActionSplice(Carbon::Check::Context&, Carbon::SemIR::LocIdAndInst, Carbon::SemIR::TypeId) /proc/self/cwd/toolchain/check/action.cpp:146:17
```

Reply:
Well, RefineOperands is buggy:
```
  auto arg1 = RefineOperand(context, loc_id, arg0_kind, action.arg1());
```

Reply:
With the fix, `None` seems to work, but with an error. Does this seem reasonable given the current state of affairs?

Reply:
Yeah, I think that's fine; my concern was that `None` is a nullish thing, and we'd be passing it through logic that hadn't seen a nullish thing before. If that already works enough to not crash, that's probably OK.